Sources/Target/Common/ThreadBase.cpp

```diff
@@ -30,5 +30,17 @@ ErrorCode ThreadBase::modifyRegisters(
   action(state);
   return writeCPUState(state);
 }
+
+ErrorCode ThreadBase::beforeResume() {
+  if (!_process->isAlive())
+    return kErrorProcessNotFound;
+
+  BreakpointManager *bpm = _process->hardwareBreakpointManager();
+  if (bpm != nullptr) {
+    bpm->enable();
```

Comment:
Oh that's nice. If gdb has support for it, trying to make this clean and
ready for us to support it too makes sense.

Thanks for taking care of the cleanups :p

On Tuesday, August 9, 2016, Francis Ricci <notifications@github.com> wrote:

> In Sources/Target/Common/ThreadBase.cpp
> <https://github.com/facebook/ds2/pull/243#discussion_r74103163>:
>
> > @@ -30,5 +30,17 @@ ErrorCode ThreadBase::modifyRegisters(
> >    action(state);
> >    return writeCPUState(state);
> >  }
> > +
> > +ErrorCode ThreadBase::beforeResume() {
> > +  if (!_process->isAlive())
> > +    return kErrorProcessNotFound;
> > +
> > +  BreakpointManager *bpm = _process->hardwareBreakpointManager();
> > +  if (bpm != nullptr) {
> > +    bpm->enable();
>
> oooo, I see. Also, I looked and gdb *does* support them. So not
> impossible for it to be someday useful. I'll try to do it with minimal
> grossness.
>
> —
> You are receiving this because you modified the open/close state.
> Reply to this email directly, view it on GitHub
> <https://github.com/facebook/ds2/pull/243/files/2a73b10db4691d10fd51d07ec46fbded7b36f6af..b32f1128fa8ceb0010cf5f79262d062aca477162#r74103163>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AADHbN3iCObvqDjuHlVs20YrfEv3pM6pks5qeLb2gaJpZM4H9FRy>
> .
>


-- 
Stephane Sezer